src/main/java/emu/grasscutter/server/packet/recv/HandlerGetGachaInfoReq.java

```diff
@@ -15,6 +18,13 @@ public void handle(GameSession session, byte[] header, byte[] payload) throws Ex
 					// TODO: use other Nonce/key insteadof session key to ensure the overall security for the player
 					session.getPlayer().getAccount().getSessionKey())
 					);
+
+		for(GachaBanner banner : session.getServer().getGachaManager().getGachaBanners().values()) {
+			if(!banner.hasEpitomized()) continue;
+
+			PlayerGachaBannerInfo gachaInfo = session.getPlayer().getGachaInfo().getBannerInfo(banner);
+			session.send(new PacketGachaWishRsp(banner.getGachaType(), banner.getScheduleId(), gachaInfo.getWishItemId(), gachaInfo.getFailedFeaturedItemPulls(5), banner.getWishMaxProgress()));
```

Comment:
This seems strangely placed here. We don't usually send an *Rsp without a preceeding matching *Req. This should probably only be sent in response to GachaWishReq, including the information about failed pulls (which is currently always set to zero there).

Reply:
If dont send Rsp on this, when click to weapon banner, it dont show current wished progress

Reply:
GachaWishReq only called when click Chart Course or Cancel Course, i set wish progress to zero because when Cancel Course will reset current progress, when click to banner it only call GetGachaInfoRsp and will not show current wish progress, i add GachaWishRsp in GetGachaInfoRsp will make client see current wish progress, watch my video below
[HandlerGetGachaInfoReq with GachaWishRsp.zip](https://github.com/Grasscutters/Grasscutter/files/8898556/HandlerGetGachaInfoReq.with.GachaWishRsp.zip)
[HandlerGetGachaInfoReq without GachaWishRsp.zip](https://github.com/Grasscutters/Grasscutter/files/8898557/HandlerGetGachaInfoReq.without.GachaWishRsp.zip)

Reply:
Ah, yes, because `wish_item_id`, `wish_progress` and `wish_max_progress` are not set in `GetGachaInfoRsp`. Have a look at `GachaBanner::toProto(String)`.

Reply:
i have updated